pkg/datastore/test/caveat.go

```diff
@@ -0,0 +1,42 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/authzed/spicedb/internal/testfixtures"
+	"github.com/authzed/spicedb/pkg/datastore"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+	"github.com/authzed/spicedb/pkg/tuple"
+
+	"github.com/stretchr/testify/require"
+)
+
+// SimpleCaveatTest makes sure a caveated tuples is stored and retrieved from the datastore correctly
+func SimpleCaveatTest(t *testing.T, tester DatastoreTester) {
```

Comment:
Can you make this a table driven test that exercises nil, empty, small (current), and large caveat values?

Reply:
@jakedt good catch, addressed in https://github.com/authzed/spicedb/pull/773/commits/9e7b75c429ce83769520e6443e19bbdda2ba82fe.

Also identified datastores weren't validating protos, as the assumption was gRPC controllers would do this, but does not hold true anymore as we now use `core` for `WriteRelationships`. Added test in https://github.com/authzed/spicedb/pull/773/commits/e93451c38cd69e592614fefb9000d84c2939be1a